src/App.svelte

```diff
@@ -20,19 +20,35 @@
   let mapState = {};
   let maps = [];
   const config = makeConfig(localConfig);
-  let settings = getInitialSettings(config);
+  let settings = getSettings(config);
   let mapboxGlAccessToken;
   let stylePresetUrls;
   configStore.set(config);
   ({ mapboxGlAccessToken, stylePresetUrls } = config);
 
+  let settingsForHash = {};
+  $: if (settings || maps || mapState) {
+    // Remove the stylesheet for a more concise hash
+    const mapsHash = JSON.parse(JSON.stringify(maps)).map(m => {
+      delete m.style;
+      return m;
+    });
+
+    settingsForHash = { ...settings, maps: mapsHash, ...mapState };
+  }
+
+  // Detect changes in hash and update settings appropriately
+  window.addEventListener('hashchange', () => {
+    if (location.hash.slice(1) !== getHashString(settingsForHash)) {
+      settings = getSettings(config);
+    }
+  });
```

Comment:
If the current hash is different from what the hash changed to, update the app settings.